### PR TITLE
UI for create category

### PR DESCRIPTION
**Pull request by pittalamadhuri**

### CODE REVIEW

File: src/components/Select/SelectComponent.tsx [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
     list: any[];
     label: string;
-    // handleChange: () => void;
+    handleChange: (value: string) => void;
 }
 
 export function SelectComponent(props: Props) {
-    const [selected, updateSelected] = React.useState();
 
     function handleChange(e: any){
```
Comment (saiumesh535) [inline src/components/Select/SelectComponent.tsx:16]: avoid any

File: src/components/TextComponent/TextComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+interface Props {
+    label: String;
+    handler: (target: any, formElement: string| undefined) => void;
+    formElement?: string;
+}
+
+function TextComponent( props: Props ): JSX.Element {
+    return (
+        <TextField
```
Comment (saiumesh535) [inline src/components/TextComponent/TextComponent.tsx:11]: Indentation

File: src/components/TextComponent/TextComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { TextField } from '@mui/material';
+
+interface Props {
+    label: String;
+    handler: (target: any, formElement: string| undefined) => void;
```
Comment (saiumesh535) [inline src/components/TextComponent/TextComponent.tsx:5]: avoid any

File: src/data/sagas.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import { createCategoryAPI, createExpenseAPI, getCategoriesAPI } from './backend';
+import { Action, ZustandState } from './store';
+
+function* createCategory(input: Action<Category>) {
+    const categories: Category[] = yield select((state: ZustandState) => state.categories);
+    try {
+        yield createCategoryAPI(input.payload);
+        yield setState({ categories: [...categories, input.payload] });
+    } catch (e) {
+        console.error("Error in saga");
```
Comment (saiumesh535) [inline src/data/sagas.ts:15]: handle errors and show to user

**Review by saiumesh535 — CHANGES_REQUESTED** [review]
Requested changes

File: src/data/sagas.ts (hunk 4 above)
Comment (pittalamadhuri) [reply src/data/sagas.ts:15]: Will do later as snackbar

File: src/components/ButtonComponent/ButtonComponent.scss [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Button {
```
Comment (saiumesh535) [inline src/components/ButtonComponent/ButtonComponent.scss:1]: avoid HTML element to change style rather use a class

File: src/pages/Category/CreateCategory.tsx [hunk 6]
```diff
@@ -1,22 +1,62 @@
...
+                        variant="outlined"
+                        value={categoryName}
+                        onChange={(e) => setCategoryName(e.target.value)}
+                    />
+                    <ButtonComponent handler={saveCategory} label="Create Category" />
+                    <Typography sx={{ mt: 4, mb: 2 }} variant="h6" component="div">
+                        Categories
+                    </Typography>
+                    <List>
+                        {categories && categories.map(category => {
```
Comment (saiumesh535) [inline src/pages/Category/CreateCategory.tsx:44]: categories?.map

**Review by saiumesh535 — APPROVED** [review]
LGTM!

**Pull request merged**